src/renderer/services/file-management-system/test/file-management-system.test.ts

```diff
@@ -82,6 +82,74 @@ describe("FileManagementSystem", () => {
   });
 
   describe("upload", () => {
+
+    it("Inits chunked upload instead of local_nas_shortcut", async () => {
+      // Arrange
+      const upload: UploadJob = {
+        ...mockJob,
+        serviceFields: {
+          files: [
+            {
+              file: {
+                fileType: "text",
+                originalPath: testFilePath,
+              },
+            },
+          ],
+          type: "upload",
+          localNasShortcut: false
+        },
+      };
+      const uploadId = "091234124";
+      const expectedMd5 = "testMd5";
+      fss.fileExistsByNameAndSize.resolves(false);
+      fss.registerUpload.resolves({ status: UploadStatus.WORKING, uploadId, chunkSize: 2424, chunkStatuses: [], currentFileSize: -1, fileSize: -1 });
+      fileReader.read.resolves(expectedMd5)
+
+      // Act
+      await fms.upload(upload, noop);
+
+      // Assert
+      expect(fileReader.read).to.have.been.calledOnce;
+      expect(fss.finalize.calledOnceWithExactly(uploadId, expectedMd5)).to.be.true;
+    });
+
+    it("Inits local_nas_shortcut instead of chunked.", async () => {
```

Comment:
I think a test where fssStatus is returned as `INACTIVE` and/or `RETRY` to exercise an error case would be useful